Controller/StatusController.php

```diff
@@ -36,8 +40,14 @@ class StatusController extends FOSRestController
      */
     public function getStatusAction()
     {
-        $data = array('OK','0', 'All related services are available. All systems normal.');
-
-        return $this->view($data, 200);
+        $dispatcher = $this->get('event_dispatcher');
+        $event = new StatusCheckEvent();
+        $dispatcher->dispatch(Events::STATUS_CHECK, $event);
+        $messages = $event->getMessages();
+        if (count($messages)) {
```

Comment:
we should add a "state" property to the event and check that instead of counting the messages.  Default state can be "ok" and can be set to "warning" or "error" by any listener.  A listener should not be able to "downgrade" state from "error" to "warning" so the setState method should only allow escalating the state and not downgrading it.